### PR TITLE
Quicksort

### PR DESCRIPTION
**Pull request by Villiamfj**
- QuickSort benchmark, using the same input as MergeSort
- Found and fixed some issues in the MergeSort benchmarks

### CODE REVIEW

File: benchmarks/quicksort/c/bench.c
Comment (Mads518d) [inline benchmarks/quicksort/c/bench.c:1]: Der bliver brugt en del ref til merge hvilket nok ikke er helt korrekt

File: benchRunners/quicksort.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
+Input=`cat benchRunners/mergeSortParam` # getting input from file
+count=1000
```
Comment (Jakobzs) [inline benchRunners/quicksort.sh]: Count skal være under "testname" og "folder" reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

**Review by Jakobzs — CHANGES_REQUESTED** [review]

File: benchmarks/quicksort/c/bench.c
Comment (Jakobzs) [reply benchmarks/quicksort/c/bench.c:1]: SwagGIF

File: benchmarks/quicksort/cpp/bench.cpp
Comment (Mads518d) [inline benchmarks/quicksort/cpp/bench.cpp:1]: Der bliver også her ref'et til merge

File: benchmarks/quicksort/java/Bench.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        
+        // Getting arguments
+        // converting json array to java array
+        String[] data = args[1].replace("[","").replace("]","").split(",");
+        List<Long> mergeParam = Arrays.stream(data).map(String::trim).map(Long::valueOf).toList();
+        int loop_count = Integer.parseInt(args[0]);
+
+        // Running benchmark
+        // Note that this could potentially be optimized away
+        // by the compiler due to the fact that the result is not used.
```
Comment (Mads518d) [inline benchmarks/quicksort/java/Bench.java]:
```suggestion
```

File: benchmarks/quicksort/java/Bench.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        
+        // Getting arguments
+        // converting json array to java array
+        String[] data = args[1].replace("[","").replace("]","").split(",");
+        List<Long> mergeParam = Arrays.stream(data).map(String::trim).map(Long::valueOf).toList();
+        int loop_count = Integer.parseInt(args[0]);
+
+        // Running benchmark
+        // Note that this could potentially be optimized away
```
Comment (Mads518d) [inline benchmarks/quicksort/java/Bench.java]:
```suggestion
```

File: benchmarks/quicksort/python/bench.py [hunk 4]
```diff
@@ -0,0 +1,50 @@
+# NOTE MUST BE CALLED FROM ROOT
+
+from ctypes import *
+import sys
+import platform
+
+# used in test method
+from heapq import merge
```
Comment (Mads518d) [inline benchmarks/quicksort/python/bench.py]: ikke brugt

File: benchRunners/quicksort.sh (hunk 1 above)
Comment (Villiamfj) [reply benchRunners/quicksort.sh]: IAmDoneGIF

File: benchRunners/quicksort.sh [hunk 5]
```diff
@@ -0,0 +1,66 @@
+testName="quicksort"
+folder="quicksort"
+count=1000
+Input=`cat benchRunners/mergeSortParam` # getting input from file
```
Comment (Jakobzs) [inline benchRunners/quicksort.sh]:
```suggestion
input=`cat benchRunners/mergeSortParam` # getting input from file
```

Comment (Jakobzs) [reply benchRunners/quicksort.sh]: Lille i reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

**Review by Jakobzs — CHANGES_REQUESTED** [review]

File: benchRunners/quicksort.sh (hunk 5 above)
Comment (Mads518d) [reply benchRunners/quicksort.sh:4]: 🤓

Comment (Mads518d) [reply benchRunners/quicksort.sh]: ಠ_ಠ

Comment (Villiamfj) [reply benchRunners/quicksort.sh]: Lavede en ctrl f

File: benchmarks/mergesort/javascript/bench.js [hunk 6]
```diff
@@ -49,14 +49,14 @@ const stop = lib.func('void stop_rapl()');
 // running benchmark
 for (let i = 0; i < runCount; i++) {
 
-    let tobeSorted = Array.from(mergeParam);
+    let toBeSorted = Array.from(mergeParam);
```
Comment (Mads518d) [inline benchmarks/mergesort/javascript/bench.js:52]: Du kan meget godt lide denne "mergeParam"

**Review by Mads518d — APPROVED** [review]
if it works it works

**Review by Jakobzs — APPROVED** [review]

**Pull request merged**